sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1724,6 +1724,10 @@ object PushPredicateThroughNonJoin extends Rule[LogicalPlan] with PredicateHelpe
       val aliasMap = getAliasMap(project)
       project.copy(child = Filter(replaceAlias(condition, aliasMap), grandChild))
 
+    // We can push down deterministic predicate through Aggregate, including throwable predicate.
+    // If we can push down a filter through Aggregate, it means the filter only references the
+    // grouping keys. The Aggregate operator can't reduce grouping keys so the filter won't see any
```

Comment:
```suggestion
    // grouping keys or constants. The Aggregate operator can't reduce distinct values of grouping keys so the filter won't see any
```

Reply:
why need "or constants" when `replaced.references.subsetOf(aggregate.child.outputSet)`?

Reply:
e.g. `Filter(a > 1, Aggregate(1 as a, max(col), ...))`, we can push down the filter, right?

Reply:
yea, updated.